src/main/java/com/faforever/api/user/GogService.java

```diff
@@ -108,11 +108,14 @@ private GogGamesListPage getGamesListPage(String gogUsername, int page) {
     try {
       return restTemplate.getForObject(gamesListUrl, GogGamesListPage.class);
     } catch (HttpClientErrorException e) {
-      throw switch (e.getStatusCode()) {
-        case FORBIDDEN -> ApiException.of(ErrorCode.GOG_LINK_GAMES_NOT_PUBLIC);
-        case NOT_FOUND -> ApiException.of(ErrorCode.GOG_LINK_USER_NOT_FOUND);
-        default -> e;
-      };
+      switch ((HttpStatus) e.getStatusCode()) {
+        case FORBIDDEN:
+          throw ApiException.of(ErrorCode.GOG_LINK_GAMES_NOT_PUBLIC);
+        case NOT_FOUND:
+          throw ApiException.of(ErrorCode.GOG_LINK_USER_NOT_FOUND);
+        default:
+          throw e;
+      }
```

Comment:
Please use new switch expressions with `->`

Reply:
it's not working. feel free to try. it's sealed interface now and looks like it isn't that much supported with java 17. If I've enabled preview features, it does not complain (gradle), but idea refuses to compile

Reply:
Ah ok. Well then for the time bein its fine.